src/components/TaskDetailsModal/TaskDetailsModal.js

```diff
@@ -40,61 +39,70 @@ class TaskDetailsModal extends PureComponent<Props> {
     isRunning ? abortTask(task, timestamp) : runTask(task, timestamp);
   };
 
-  getStatusText = () => {
-    const { status, timeSinceStatusChange } = this.props.task;
+  renderPrimaryStatusText = () => {
+    const { status } = this.props.task;
 
     switch (status) {
-      case 'idle': {
+      case 'idle':
         return (
-          <span>
+          <PrimaryStatusText>
             Task is <strong>idle</strong>.
-            {timeSinceStatusChange && (
-              <LastRunText>
-                Last run:{' '}
-                {moment(timeSinceStatusChange).format(
-                  'MMMM Do, YYYY [at] h:mm a'
-                )}
-              </LastRunText>
-            )}
-          </span>
+          </PrimaryStatusText>
         );
-      }
-      case 'pending': {
+
+      case 'pending':
         return (
-          <span>
+          <PrimaryStatusText>
             Task is{' '}
             <strong style={{ color: COLORS.orange[500] }}>running</strong>...
-          </span>
+          </PrimaryStatusText>
         );
-      }
 
-      case 'success': {
+      case 'success':
         return (
-          <span>
+          <PrimaryStatusText>
             Task{' '}
             <strong style={{ color: COLORS.green[700] }}>
               completed successfully
             </strong>.
-            <LastRunText>
-              {moment(timeSinceStatusChange).calendar()}
-            </LastRunText>
-          </span>
+          </PrimaryStatusText>
         );
-      }
 
-      case 'failed': {
+      case 'failed':
         return (
-          <span>
+          <PrimaryStatusText>
             Task <strong>failed</strong>.
-            <LastRunText>
-              {moment(timeSinceStatusChange).calendar()}
-            </LastRunText>
-          </span>
+          </PrimaryStatusText>
+        );
+
+      default:
+        throw new Error('Unrecognized status in TaskDetailsModal.');
+    }
+  };
+
+  renderTimestamp = () => {
+    const { status, timeSinceStatusChange } = this.props.task;
+
+    switch (status) {
+      case 'idle':
+        return (
+          <LastRunText>
+            Last run:{' '}
+            {moment(timeSinceStatusChange).format('MMMM Do, YYYY [at] h:mm a')}
```

Comment:
Could do something like

```js
{timeSinceStatusChange ? moment(timeSinceStatusChange).format('MMMM Do, YYYY [at] h:mm a') : 'Never'}
```

Reply:
Good catch! 

What do you think about just returning `null` before the `switch` if `timeSinceStatusChange` is `undefined`? I think I prefer that, since I'm not sure there's value in seeing "Last run: Never".

Reply:
👍 Sounds good, so the 'Last Run' info would only display if `timeSinceStatusChange` actually exists?

Reply:
Yep! I think that logic makes sense. That way the UI is nice and clear when you first see a task, but it'll show up when there's a history to tell the user about.